src/NinjaTools.FluentMockServer.API/Extensions/HttpRequestExtensions.cs

```diff
@@ -0,0 +1,39 @@
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Http;
+
+namespace NinjaTools.FluentMockServer.API.Extensions
+{
+    public static class HttpRequestExtensions
+    {
+
+        /// <summary>
+        /// Retrieve the raw body as a string from the Request.Body stream
+        /// </summary>
+        /// <param name="request">Request instance to apply to</param>
+        /// <param name="encoding">Optional - Encoding, defaults to UTF8</param>
+        /// <returns></returns>
+        public static async Task<string> GetRawBodyStringAsync([NotNull] this HttpRequest request, Encoding encoding = null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the overloading mechanism instead of the optional parameters.](https://app.codacy.com/app/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4722826)